runtime/src/main/java/io/quarkus/jsch/JSchSessions.java

```diff
@@ -0,0 +1,157 @@
+package io.quarkus.jsch;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import jakarta.annotation.PreDestroy;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.ProxyHTTP;
+import com.jcraft.jsch.Session;
+import com.jcraft.jsch.Slf4jLogger;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.jsch.runtime.JSchRuntimeConfig;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.configuration.ConfigurationException;
+import io.smallrye.config.SmallRyeConfig;
+
+public class JSchSessions {
+
+    private final ConcurrentHashMap<String, Session> sessions = new ConcurrentHashMap<>();
+
+    public JSchSessions() {
+        JSch.setLogger(new Slf4jLogger());
+    }
+
+    /**
+     * Get a session by name
+     *
+     * @param sessionName the session name
+     * @return the session
+     */
+    public static Session fromName(String sessionName) {
+        return Arc.container().instance(JSchSessions.class).get().getSession(sessionName);
+    }
+
+    /**
+     * Disconnect all sessions
+     */
+    public static void shutdown() {
+        Arc.container().instance(JSchSessions.class).get().disconnect();
+    }
+
+    /**
+     * Get or create a session by name
+     *
+     * @param sessionName the session name
+     * @return the session
+     */
+    public Session getSession(String sessionName) {
+        return sessions.computeIfAbsent(sessionName, this::createSession);
```

Comment:
Can you consume `Session` concurrently? Not sure they are thread-safe

Reply:
`sessions` in a `ConcurrentHashMap`.
Is it enough or not ?

Reply:
Not really, you can still have two threads using the same sessions. A ConcurrentHashMap only ensures that no two sessions of the same name are created when requested

Reply:
and if I add `synchronized` to the method `private synchronized Session createSession(String sessionName)` ?

Reply:
It can still be reused by 2 or more threads. As I said above, the ConcurrentHashMap already takes care of creating a single Session for a single name, but when consumers request it, they will synchronize only during the session creation, what happens next is that the same session is used in every consumer running in separate threads

Reply:
If I rollback to the `RequestScoped`, the Session injected will be threadSafe. Right ?

Reply:
Yeah, that may work I think.